### PR TITLE
BCIP 103: continuous blocksize increase up to 3MB

### PR DESCRIPTION
**Pull request by orangeman**
**Alternative proposal to pull request** https://github.com/bitcoinclassic/bitcoinclassic/pull/5
- increase by 30 bytes per block
- stop at max blocksize 3MB

### CODE REVIEW

File: src/consensus/consensus.h [hunk 1]
```diff
@@ -6,10 +6,36 @@
...
+    if (nTime < BCIP01_FORK_TIME)
+        return 1000*1000;
+
+    // cap for tests
+    if (nTime > 4113158400)
+        nTime = 4113158400;
+
+/** Smooth transition of 30 bytes every 10 minutes, until 3 megabytes is reached */
+    if (((2*1000*1000) + (30 * ((nTime - BCIP01_FORK_TIME) / 600))) <= 3000000) {
+      return (2*1000*1000) + (30 * ((nTime - BCIP01_FORK_TIME) / 600));
```
Comment (orangeman) [inline src/consensus/consensus.h:24]: Segregated Witness might be helpful here
Do we have a decision on https://github.com/bitcoinclassic/bitcoinclassic/pull/8 yet?

File: src/consensus/consensus.h [hunk 2]
```diff
@@ -6,10 +6,36 @@
...
+    }
+    if (((2*1000*1000) + (30 * ((nTime - BCIP01_FORK_TIME) / 600))) > 3000000){
+      return 4000000;
```
Comment (murchandamus) [inline src/consensus/consensus.h:27]: Judging from the title of your PR, line 27 should `return 3000000`?

**Comment by murchandamus** [comment]
Currently NACK: Consensus is a single increase to 2MB.

File: src/consensus/consensus.h [hunk 3]
```diff
@@ -6,10 +6,36 @@
 #ifndef BITCOIN_CONSENSUS_CONSENSUS_H
 #define BITCOIN_CONSENSUS_CONSENSUS_H
 
+#include <stdint.h>
+
+static const uint64_t BCIP103_FORK_TIME = 1456790400; // April 01 2016, midnight UTC
```
Comment (ajtowns) [inline src/consensus/consensus.h:11]: That's March 1st 2016, not April 1st... April 1st midnight should be 1459468800

**Comment by ahmedbodi** [comment]
Thanks for the PR @orangeman , the final agreement based on the data on bitcoinclassic.consider.it has settled on a 2MB upgrade for the first release, so i'll close this one in favour of that

Links:

https://bitcoinclassic.consider.it/merge-3-v0112-2mb-4mb?results=true
https://bitcoinclassic.consider.it/fork-to-2-mb-and-thats-all?results=true
https://bitcoinclassic.consider.it/fork-to-2-mb-with-scaling-up-to-4-mb-over-2-years?results=true Ahmed

**Pull request closed without merging**